### ISSUE TITLE
Too much printed feedback

### ISSUE THREAD

**Issue by swkeemink** [rochefort-lab/fissa#155]
Every time FISSA is run you get a per-cell report about the number of iterations and whether NMF converged. This is not useful in most cases and clutters the code-feecback. Instead, I would like to update it so that FISSA just reports 'all converged' if all is well, and a warning message like 'ROIs 1, 2, and 5 did not converge, consider increasing the number of iterations if this happens for a significant number of cells.'

**Comment by scottclowe** [comment]
Need to add a verbosity parameter to control this.

**Comment by swkeemink** [comment]
Yes, that as well.
I would suggest that with verbose on, the output is as it is now.
Then with verbose off, the output just puts the main things (like 'finished prep', 'finished sep', and 'ROIs 1,2,3 did not converge'.

**Comment by scottclowe** [comment]
No, it is better to have verbosity be an int instead of a bool, and then the user can select from about 4 verbosity levels.

0. Print nothing.
1. Show high-level progress bar using tqdm `tqdm.tqdm(pool.imap(func, iterable), total=n_images)` and high level summary (total duration, etc) at the end, summarising the output over all images/tiffs.
2. Show one or two lines of what is going on within each of the high level steps (in our case, a one or two line output for each image/tiff). Note that you don't show the tqdm progress here.
3. Show more detailed progress (what is going on within the processing of each image) - i.e. a one line statement for each cell being processed.
4. Could go even more detailed and show progress within processing each cell.

The default level is usually 1 or 2.

It is for the most part just as easy to code a int verbosity parameter as a bool verbose parameter. Just do
```python
if verbosity >= 1:
    print("thing")
```
instead of
```python
if verbose:
    print("thing")
```
The only thing which is a little involved is checking that `verbosity == 1` when deciding to do the tqdm progress bar.

**Comment by swkeemink** [comment]
I've implemented the above in this branch: https://github.com/swkeemink/fissa/tree/add_verbosity
But still need to get the progress bar working (not worked with tqdm before). After that is added will make a PR.

**Comment by scottclowe** [comment]
It is quite common to have child functions you are calling and want them to be less verbose than the parent. The way I do this is have the parent call the child with `fn(..., verbose=verbose-1)`. That way the child can have sensible verbosity levels (silent at verbose=0, printing things at verbose=1, more things at verbose=2), but be quieter than the parent.

It doesn't make sense to have a function which is silent for `0<verbose<3`.

**Comment by swkeemink** [comment]
This has now been implemented by #200.
